### PR TITLE
fix(learn): add test to return early pattern

### LINKED ISSUE #38750: Missing test case in Basic JavaScript: Return Early Pattern for Functions
**Describe your problem and how to reproduce it:**
***Problem description in challenge***: 
`Modify the function abTest so that if a or b are less than 0 the function will immediately exit with a value of undefined.`

***Issue***:
Both of the following solutions result in passing the challenge
``` javascript
function abTest(a, b) {
  // Only change code below this line

    if(a<=0 || b<=0)
      return undefined;

  // Only change code above this line

  return Math.round(Math.pow(Math.sqrt(a) + Math.sqrt(b), 2));
}
```
```javascript
function abTest(a, b) {
  // Only change code below this line

    if(a<0 || b<0)
      return undefined;

  // Only change code above this line

  return Math.round(Math.pow(Math.sqrt(a) + Math.sqrt(b), 2));
}
```

Based on the problem description, I believe that only the second solution should be correct.
An additional test case with either a or b being equal to 0 should be added to this challenge

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions

**Tell us about your browser and operating system:**
* Browser Name: Google Chrome
* Browser Version: Version 81.0.4044.122 (Official Build) (64-bit)
* Operating System: macOS Version 10.14.6

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by ERElli**
Currently no test for a = 0 or b = 0. This results in if(a<=0 || b<= 0) also being a valid solution

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38750

### CODE REVIEW

**Comment by raisedadead** [comment]
Hi @ERElli 

Thanks for the quick PR.

We would need a third party confirmation on the issue thread (or anywhere else, like forum) before we can land this. Keeping this blocked for the moment.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.english.md [hunk 1]
```diff
@@ -46,7 +46,9 @@ tests:
     testString: assert(abTest(2,8) === 18 );
   - text: <code>abTest(3,3)</code> should return <code>12</code>
     testString: assert(abTest(3,3) === 12 );
-
+  - text: <code>abTest(0,2)</code> should return <code>2</code>
+    testString: assert(abTest(0,2) === 2);
```
Comment (Sky020) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.english.md:50]:
```suggestion
  - text: <code>abTest(0,0)</code> should return <code>0</code>
    testString: assert(abTest(0,0) === 0);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Change the test case, to avoid this code from passing, as well:
```
if (a < 0 || b <= 0) {
  return undefined;
}
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.english.md (hunk 1 above)
Comment (ERElli) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.english.md]: Good catch! Fixed at a5c9ad9

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by hassaanp — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @ERElli 🎉 Congratulations on your first contribution to freeCodeCamp 🎉 

For future reference, it's good to give your branch a more descriptive name and start it with a scope. See `#3` just below [here in the docs](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally.md#making-changes-locally)

**Comment by ERElli** [comment]
Thanks @moT01!

I realized that I hadn't named my branch properly after I had made the PR and there were already comments. I wasn't sure if closing the PR and renaming the branch was the right thing to do.

I'll make sure to name my branch properly next time!

**Pull request merged**